### PR TITLE
Add mill cert example

### PR DESCRIPTION
**Pull request by OR13**
- Added example vc-json-schema document for steel mill cert test report
- Added example vc for steel mill cert test report which uses the schema
- Added test for signing the the schema with jsonld
- Added test for issuing/verifying the steel mill cert test report vc
- Added test for verifying the unsigned mill cert vc matches the vc-json-schema

### CODE REVIEW

**Comment by OR13** [comment]
Sorry this is one commit :(

I spent a lot of time sanity checking jsonld issues, and chasing documentLoader bugs... still 35 changes in 1 commit is pretty sad.

File: docs/schema/steel-mill-cert-test-report.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{
+  "@context": [
+    "https://w3id.org/did/v1",
+    "https://w3id.org/security/v2",
+    "https://w3c-ccg.github.io/vc-json-schemas/context/vc-json-schema-v0.0.jsonld"
+  ],
+  "type": "https://w3c-ccg.github.io/vc-json-schemas/schema/credential-schema.json",
+  "modelVersion": "1.0",
+  "id": "https://w3c-ccg.github.io/vc-json-schemas/schema/steel-mill-cert-test-report.json",
```
Comment (glcohen) [inline docs/schema/steel-mill-cert-test-report.json]: should be a valid identifier according to https://w3c-ccg.github.io/vc-json-schemas/#id

I guess, given the vague language of the draft this is compliant...

File: hypothetical-use-cases/trade-certificates/examples/steel-mill-cert-test-report/unsigned.schema.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+  "@context": [
+    "https://w3id.org/did/v1",
+    "https://w3id.org/security/v2",
+    "https://w3c-ccg.github.io/vc-json-schemas/context/vc-json-schema-v0.0.jsonld"
+  ],
+  "type": "https://w3c-ccg.github.io/vc-json-schemas/schema/credential-schema.json",
+  "modelVersion": "1.0",
+  "id": "https://w3c-ccg.github.io/vc-json-schemas/schema/steel-mill-cert-test-report.json",
```
Comment (glcohen) [inline hypothetical-use-cases/trade-certificates/examples/steel-mill-cert-test-report/unsigned.schema.json]: same here

File: hypothetical-use-cases/trade-certificates/examples/steel-mill-cert-test-report/unsigned.vc.json [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://w3c-ccg.github.io/vc-json-schemas/context/hypothetical-examples/trade-certificates-v0.0.jsonld"
+  ],
+  "id": "https://w3c-ccg.github.io/vc-json-schemas/credentials/steel-mill-cert-test-report.example.vc.jsonld",
+  "type": ["VerifiableCredential", "CertifiedMillTestReport"],
+  "issuer": "did:key:z6MksUNTUBMbQTzpwAnQ6VFp7aVXTEi7gZbAwpfXkftwDKzK",
+  "issuanceDate": "2019-12-02T15:08:37.751Z",
+  "credentialSchema": {
+    "id": "https://w3c-ccg.github.io/vc-json-schemas/schema/steel-mill-cert-test-report.json",
```
Comment (glcohen) [inline hypothetical-use-cases/trade-certificates/examples/steel-mill-cert-test-report/unsigned.vc.json]: and here

File: docs/schema/steel-mill-cert-test-report.json (hunk 1 above)
Comment (OR13) [reply docs/schema/steel-mill-cert-test-report.json]: yes, i knew we would need to discuss this. IMO, we should not require this to be a DID URI, for many reasons, most of which are discussed on this thread:

https://github.com/hyperledger/aries-rfcs/issues/129

IMO, we should take this opportunity to document the discussion that happened in aries, and make it clear that the consumer has some choice in spec compliance here.

I will create an issue to track this.

Comment (OR13) [reply docs/schema/steel-mill-cert-test-report.json]: https://github.com/w3c-ccg/vc-json-schemas/issues/17

File: hypothetical-use-cases/trade-certificates/examples/steel-mill-cert-test-report/unsigned.schema.json (hunk 2 above)
Comment (OR13) [reply hypothetical-use-cases/trade-certificates/examples/steel-mill-cert-test-report/unsigned.schema.json]: https://github.com/w3c-ccg/vc-json-schemas/issues/17

File: hypothetical-use-cases/trade-certificates/examples/steel-mill-cert-test-report/unsigned.vc.json (hunk 3 above)
Comment (OR13) [reply hypothetical-use-cases/trade-certificates/examples/steel-mill-cert-test-report/unsigned.vc.json]: https://github.com/w3c-ccg/vc-json-schemas/issues/17

File: docs/schema/steel-mill-cert-test-report.json (hunk 1 above)
Comment (OR13) [reply docs/schema/steel-mill-cert-test-report.json]: Can I mark as resolved?

Comment (glcohen) [reply docs/schema/steel-mill-cert-test-report.json]: yes

**Comment by glcohen** [comment]
I am not qualified to review javascript, but the rest looks good

**Comment by OR13** [comment]
I'm gonna gather a bit more feedback from team / community before merging this.

File: hypothetical-use-cases/trade-certificates/examples/resolver.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const didDocs = require("./didDocs");
```
Comment (gjgd) [inline hypothetical-use-cases/trade-certificates/examples/resolver.js]: duplicate of `hypothetical-use-cases/basic/resolver.js`

File: hypothetical-use-cases/basic/didDocs.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      type: "X25519KeyAgreementKey2019",
+      controller: "did:key:z6MkfXT3gnptvkda3fmLVKHdJrm8gUnmx3faSd1iVeCAxYtP",
+      publicKeyBase58: "DnyBg4MYpNdF4JQihynmCiC5hpiZh5tkQdx4QTfk1xE5"
+    }
+  ]
+};
+
+const docs = [alice];
+
+docs.map(didDoc => {
```
Comment (gjgd) [inline hypothetical-use-cases/basic/didDocs.js]: nit: [.forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) is more appropriate when there are side effects

**Comment by OR13** [comment]
@gjgd requires node 12

**Comment by wyc** [comment]
Tests pass on my end:
```
% node --version
v13.3.0
Test Suites: 3 passed, 3 total
Tests:       3 passed, 3 total
Snapshots:   0 total
Time:        5.113s
Ran all test suites.
```

File: hypothetical-use-cases/trade-certificates/examples/resolver.js (hunk 4 above)
Comment (OR13) [reply hypothetical-use-cases/trade-certificates/examples/resolver.js]: I have a feeling we may remove these later, for now its helpful to have the dependencies separated.

**Pull request merged**